### PR TITLE
[FEATURE][ui] Show dataset file size and last modified information in the layer properties window

### PR DESCRIPTION
**Pull request by nirvn**
I can't count the times I've opened a file explorer only wanting to know the file size of my dataset. Let's add that in our layer properties' general information section:

image

I've taken the opportunity to de-duplicate some code across several layer types.

--

For the record (and documentation), this feature also improves the browser panel's information section:
image

### CODE REVIEW

**Comment by nirvn** [comment]
(Unrelated test failure)

File: src/core/qgsmaplayer.cpp [hunk 1]
```diff
@@ -2189,6 +2191,75 @@ void QgsMapLayer::invalidateWgs84Extent()
...
+    if ( uriComponents.contains( QStringLiteral( "path" ) ) )
+    {
+      path = uriComponents[QStringLiteral( "path" )].toString();
+      QFileInfo fi( path );
+      if ( fi.exists() )
+      {
+        isLocalPath = true;
+        metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Path" ) + QStringLiteral( "</td><td>%1" ).arg( QStringLiteral( "<a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( path ).toString(), QDir::toNativeSeparators( path ) ) ) + QStringLiteral( "</td></tr>\n" );
+
+        qint64 fileSize = fi.size();
```
Comment (nyalldawson) [inline src/core/qgsmaplayer.cpp:2216]: what happens here for non-file sources? (e.g. a .gdb folder). Should the `fi.isFile()` test be moved up before this?

File: src/core/qgsmaplayer.cpp [hunk 2]
```diff
@@ -2189,6 +2191,75 @@ void QgsMapLayer::invalidateWgs84Extent()
...
+                lastModified = sidecarFi.lastModified();
+                lastModifiedFileName = sidecarFi.fileName();
+              }
+              sidecarFileNames << sidecarFi.fileName();
+            }
+            metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + ( sidecarFiles.size() > 1 ? tr( "Sidecar files" ) : tr( "Sidecar file" ) ) + QStringLiteral( "</td><td>%1" ).arg( sidecarFileNames.join( QLatin1String( ", " ) ) ) + QStringLiteral( "</td></tr>\n" );
+          }
+          metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + ( !sidecarFiles.isEmpty() ? tr( "Total size" ) : tr( "Size" ) ) + QStringLiteral( "</td><td>%1" ).arg( QgsFileUtils::representFileSize( fileSize ) ) + QStringLiteral( "</td></tr>\n" );
+        }
+        metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Last modified" ) + QStringLiteral( "</td><td>%1" ).arg( QLocale().toString( fi.lastModified() ) ) + ( !lastModifiedFileName.isEmpty() ? QStringLiteral( " (%1)" ).arg( lastModifiedFileName ) : "" ) + QStringLiteral( "</td></tr>\n" );
```
Comment (nyalldawson) [inline src/core/qgsmaplayer.cpp:2241]:
```suggestion
        metadata += QStringLiteral( "<tr><td class=\"highlight\">" ) + tr( "Last modified" ) + QStringLiteral( "</td><td>%1" ).arg( QLocale().toString( fi.lastModified() ) ) + ( !lastModifiedFileName.isEmpty() ? QStringLiteral( " (%1)" ).arg( lastModifiedFileName ) : QString() ) + QStringLiteral( "</td></tr>\n" );
```

**Comment by nyalldawson** [comment]
Great improvement, I like it a lot!!

File: src/core/qgsmaplayer.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/qgsmaplayer.cpp:2216]: The size returns 0. I guess we should avoid printing size for folder source. One could calculate size of files within folder but let's say for some reason people drag n dropped non-gdb files into their gdb folder, we'd end up with a bad total size.

**Comment by nirvn** [comment]
@nyalldawson , comments addressed.

**Review by nyalldawson — APPROVED** [review]

**Comment by andreasneumann** [comment]
Great! Thank you!

**Pull request merged**